src/components/loader/loader-square.test.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import Loader from ".";
+
+jest.mock("../../hooks/useMediaQuery", () => {
+  return jest.fn(() => true);
+});
+
+it.each([0, 1, 2])("each loader square renders as expected", (index) => {
+  render(<Loader />);
+
+  const loaderSquares = screen.getAllByTestId("loader-square");
+
+  expect(loaderSquares[index]).toBeInTheDocument();
+});
+
+// These styling tests required for coverage
+test("when `size` prop is set to large, the expected width, height, and margin styles are applied", () => {
+  render(<Loader size="large" />);
+
+  const loaderSquares = screen.getAllByTestId("loader-square");
+
+  expect(loaderSquares[0]).toHaveStyle({
+    height: "20px",
+    width: "20px",
+    marginRight: "8px",
+  });
+
+  expect(loaderSquares[1]).toHaveStyle({
+    height: "20px",
+    width: "20px",
+    marginRight: "8px",
+  });
+
+  expect(loaderSquares[2]).toHaveStyle({
+    height: "20px",
+    width: "20px",
+  });
+});
+
+// These styling tests required for coverage
+test("when `size` prop is set to small, the expected width, height, and margin styles are applied", () => {
+  render(<Loader size="small" />);
+
+  const loaderSquares = screen.getAllByTestId("loader-square");
+
+  expect(loaderSquares[0]).toHaveStyle({
+    width: "12px",
+    marginRight: "6px",
+  });
+
+  expect(loaderSquares[1]).toHaveStyle({
+    width: "12px",
+    marginRight: "6px",
+  });
+
+  expect(loaderSquares[2]).toHaveStyle({
+    width: "12px",
+  });
+});
+
+// These styling tests required for coverage
+test("when inside button, the expected white background colour is applied", () => {
+  render(<Loader isInsideButton />);
+
+  const loaderSquares = screen.getAllByTestId("loader-square");
+
+  expect(loaderSquares[0]).toHaveStyle({
+    backgroundColor: "var(--colorsUtilityYang100)",
+  });
+
+  expect(loaderSquares[1]).toHaveStyle({
+    backgroundColor: "var(--colorsUtilityYang100)",
+  });
+
+  expect(loaderSquares[2]).toHaveStyle({
+    backgroundColor: "var(--colorsUtilityYang100)",
+  });
+});
+
+// These styling tests required for coverage
+test("when inside button and `isActive` prop is false, the expected background colour is applied", () => {
+  render(<Loader isInsideButton isActive={false} />);
+
+  const loaderSquares = screen.getAllByTestId("loader-square");
+
+  expect(loaderSquares[0]).toHaveStyle({
```

Comment:
comment: I think we need to use toHaveStyleRule here as the jest DOM matcher returns false positives when testing token values